lib/phoenix/router/route.ex

```diff
@@ -50,6 +50,20 @@ defmodule Phoenix.Router.Route do
       dispatch: build_dispatch(route, binding)}
   end
 
+  @doc """
+  Used by the binding macros built in maybe_binding, to URI decode params
+  embedded in paths.
+
+  Strings are decoded directly. Lists of strings decodes each string.
+  """
+  def uri_decode_param_value(value) do
```

Comment:
It's fine because the module is `@moduledoc false` so it won't show up in documentation. Undocumented functions are treated as internal.